modules/simple_forms_api/app/services/simple_forms_api/notification_email.rb

```diff
@@ -315,25 +315,30 @@ def form20_10206_contact_info
 
     # email and first name for form 20-10207
     def form20_10207_contact_info
-      preparer_types = %w[veteran third-party-veteran non-veteran third-party-non-veteran]
+      return unless form_data
 
-      return unless preparer_types.include?(@form_data['preparer_type'])
+      preparer_type = form_data['preparer_type']
 
-      email_and_first_name = [@user&.va_profile_email]
-      # veteran
-      email_and_first_name << if @form_data['preparer_type'] == 'veteran'
-                                @form_data['veteran_full_name']['first']
+      return unless %w[veteran third-party-veteran non-veteran third-party-non-veteran].include?(preparer_type)
 
-                              # non-veteran
-                              elsif @form_data['preparer_type'] == 'non-veteran'
-                                @form_data['non_veteran_full_name']['first']
+      email = if preparer_type.include?('non-veteran')
+                form_data['non_veteran_email_address']
+              else
+                form_data['veteran_email_address']
+              end
 
-                                # third-party
-                              else
-                                @form_data['third_party_full_name']['first']
-                              end
+      first_name = case preparer_type
+                   when 'veteran'
+                     form_data.dig('veteran_full_name', 'first')
+                   when 'non-veteran'
+                     form_data.dig('non_veteran_full_name', 'first')
+                   when /third-party/
+                     form_data.dig('third_party_full_name', 'first')
+                   end
 
-      email_and_first_name
+      email ||= user&.va_profile_email
+
+      [email, first_name]
```

Comment:
Changes to this file shouldn't be needed since the file is deprecated. Do you mind doing a quick search in the codebase to confirm that this file is not being called from anywhere? If we're both confident it's not being called, I think we can omit this part of the diff.

Reply:
Ignore the above! Thank you thank you thank you for writing the code twice. Sorry about the paused refactor... I'll try to get it cleaned up early next week!